### PR TITLE
Make sure that the onchain privacy group version is always different.

### PR DESCRIPTION
**Pull request by pinges**
Signed-off-by: Stefan Pingel <stefan.pingel@consensys.net>

### CODE REVIEW

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/web3j/privacy/contracts/PrivacyGroupTest.java [hunk 1]
```diff
@@ -131,6 +134,30 @@ public void ensureContractIsNotLockedAfterDeploy() throws Exception {
...
+        .addParticipants(firstParticipant.raw(), Collections.singletonList(thirdParticipant.raw()))
+        .send();
+    final byte[] version2 = privacyGroup.getVersion().send();
+    privacyGroup.lock().send();
+    privacyGroup.removeParticipant(firstParticipant.raw(), secondParticipant.raw()).send();
+    final byte[] version3 = privacyGroup.getVersion().send();
+
+    System.out.println(Arrays.toString(version1));
+    System.out.println(Arrays.toString(version2));
+    System.out.println(Arrays.toString(version3));
```
Comment (lucassaldanha) [inline acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/web3j/privacy/contracts/PrivacyGroupTest.java:154]: Remove the `System.out.println` :)

File: privacy-contracts/src/main/solidity/DefaultOnChainPrivacyGroupManagementContract.sol [hunk 2]
```diff
@@ -4,11 +4,11 @@ import "./OnChainPrivacyGroupManagementInterface.sol";
 contract DefaultOnChainPrivacyGroupManagementContract is OnChainPrivacyGroupManagementInterface {
 
     bool private _canExecute;
-    int private _version;
+    bytes32 private _version;
     bytes32[] private distributionList;
     mapping(bytes32 => uint256) private distributionIndexOf;
 
-    function getVersion() external view returns (int) {
+    function getVersion() external view returns (bytes32) {
```
Comment (lucassaldanha) [inline privacy-contracts/src/main/solidity/DefaultOnChainPrivacyGroupManagementContract.sol:11]: Have you tested the impact of this change in web3j and web3js-eea? If they aren't compatible, we need at least to have GIs to update the libraries after we change this.

Comment (pinges) [reply privacy-contracts/src/main/solidity/DefaultOnChainPrivacyGroupManagementContract.sol:11]: There should not be any public APIs for the version number in web3j(s), as this is entirely handled inside Besu. But we should check that!

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/web3j/privacy/contracts/PrivacyGroupTest.java (hunk 1 above)
Comment (pinges) [reply acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/web3j/privacy/contracts/PrivacyGroupTest.java:154]: Done!

**Pull request merged**